vk_video_decoder/libs/NvVideoParser/include/VulkanAV1Decoder.h

```diff
@@ -0,0 +1,477 @@
+    uint32_t        num_ticks_per_picture;
+} av1_timing_info_t;
+
+typedef struct _av1_dec_model_info
+{
+    uint32_t        num_units_in_decoding_tick;
+    int32_t         encoder_decoder_buffer_delay_length;
+    int32_t         buffer_removal_time_length;
+    int32_t         frame_presentation_time_length;
+} av1_dec_model_info_t;
+
+typedef struct _av1_dec_model_op_params
+{
+    bool            decoder_model_param_present;
+    uint32_t        bitrate;
+    uint32_t        buffer_size;
+    int32_t         cbr_flag;
+    int32_t         decoder_buffer_delay;
+    int32_t         encoder_buffer_delay;
+    int32_t         low_delay_mode_flag;
+    int32_t         display_model_param_present;
+    int32_t         initial_display_delay;
+} av1_dec_model_op_params_t;
+
+typedef struct _GlobalMotionParams {
+    uint32_t        wmtype;
+    int32_t         wmmat[6];
+    int8_t          invalid;
+    int8_t          reserved[3];
+} GlobalMotionParams;
+
+typedef struct _av1_ref_frames_s
+{
+    VkPicIf*                buffer;
+    StdVideoAV1FrameType    frame_type;
+    StdVideoAV1FilmGrain    film_grain_params;
+    AV1WarpedMotionParams   global_models[GM_GLOBAL_MODELS_PER_FRAME];
+    int8_t                  lf_ref_delta[STD_VIDEO_AV1_NUM_REF_FRAMES];
+    int8_t                  lf_mode_delta[2];
+    bool                    showable_frame;
+    struct
+    {
+	    uint8_t				FeatureEnabled[STD_VIDEO_AV1_MAX_SEGMENTS];
+		int16_t				FeatureData[STD_VIDEO_AV1_MAX_SEGMENTS][STD_VIDEO_AV1_SEG_LVL_MAX];
+        int32_t             last_active_id;
+        uint8_t             preskip_id;
+        uint8_t             reserved[3];
+    } seg;
+
+    // Temporary variables.
+    uint32_t                primary_ref_frame; // if not 0 -- may not alloc a slot. Re-resolve this per frame per dpb index.
+    uint32_t                base_q_index;
+    bool                    disable_frame_end_update_cdf;
+    bool                    segmentation_enabled;
+
+    int8_t                  RefFrameSignBias[STD_VIDEO_AV1_NUM_REF_FRAMES];
+    uint8_t                 SavedOrderHints[STD_VIDEO_AV1_NUM_REF_FRAMES];
+    uint8_t                 order_hint;
+} av1_ref_frames_s;
+
+// AV1 decoder class
+class VulkanAV1Decoder : public VulkanVideoDecoder
+{
+protected:
+	VkSharedBaseObj<av1_seq_param_s> m_sps; // active sps
+    VkParserAv1PictureData		m_PicData;
+
+    // common params
+    int32_t                     temporal_id;
+    int32_t                     spatial_id;
+    bool                        m_bSPSReceived;
+    bool                        m_bSPSChanged;
+    bool                        m_obuAnnexB;
+    uint8_t                     timing_info_present;
+    av1_timing_info_t           timing_info;
+    av1_dec_model_info_t        buffer_model;
+    av1_dec_model_op_params_t   op_params[MAX_NUM_OPERATING_POINTS + 1];
+    uint32_t                    op_frame_timing[MAX_NUM_OPERATING_POINTS + 1];
+
+	uint8_t						delta_frame_id_length;
```

Comment:
A lot of fields are misaligned here.  I think we should use astyle or clang-format against this file.